hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileSystemDeleteHandler.java

```diff
@@ -35,21 +48,70 @@ public FileSystemDeleteHandler() {
         // Exists for injection
     }
 
+    /**
+     * This method is called during the pre-handle workflow.
+     *
+     * <p>Determines signatures needed for deleting a system file
+     *
+     * @param context the {@link PreHandleContext} which collects all information that will be
+     *     passed to {@code handle()}
+     * @throws NullPointerException if any of the arguments are {@code null}
+     */
     @Override
-    public void preHandle(@NonNull final PreHandleContext context) {
+    public void preHandle(@NonNull final PreHandleContext context) throws PreCheckException {
         requireNonNull(context);
-        throw new UnsupportedOperationException("Not implemented");
+
+        final var transactionBody = context.body().systemDeleteOrThrow();
+        final var fileStore = context.createStore(ReadableFileStoreImpl.class);
+        final var fileMeta = preValidate(transactionBody.fileID(), fileStore, context, true);
+
+        validateAndAddRequiredKeys(fileMeta.keys(), context, true);
     }
 
     /**
-     * This method is called during the handle workflow. It executes the actual transaction.
-     *
-     * <p>Please note: the method signature is just a placeholder which is most likely going to
-     * change.
+     * Given the appropriate context, deletes system file.
      *
+     * @param systemDeleteTransactionBody the {@link SystemDeleteTransactionBody} of the active
+     *     system file delete transaction
+     * @param fileStore the {@link WritableFileStoreImpl} to use to delete the file
      * @throws NullPointerException if one of the arguments is {@code null}
      */
-    public void handle() {
-        throw new UnsupportedOperationException("Not implemented");
+    public void handle(
+            @NonNull final HandleContext handleContext,
+            @NonNull final SystemDeleteTransactionBody systemDeleteTransactionBody,
+            @NonNull final WritableFileStoreImpl fileStore) {
+        requireNonNull(handleContext);
+        requireNonNull(systemDeleteTransactionBody);
+        requireNonNull(fileStore);
+
+        final var fileId = systemDeleteTransactionBody.fileIDOrElse(FileID.DEFAULT);
```

Comment:
that what I have done also in file delete and also consensus service is doing. it all will be 0 and it will fail during the validation. the other option can be that we will throw an exception for not trx in body